lib/melissa_data/web_smart/response_processor.rb

```diff
@@ -1,39 +1,40 @@
 module MelissaData
   module WebSmart
     module ResponseProcessor
-      def process_property(response)
-        codes = codes(response)
+      require 'yaml'
+      CODE_TYPES = [:property, :address]
 
+      def process(response, resp_type)
+        codes = codes(response, resp_type)
         if has_error_codes?(codes)
-          { errors: codes_for(codes, 'error') }
+          { errors: codes_for(codes, resp_type, 'error') }
         else
-          response.merge!(success: codes_for(codes, 'success'))
+          response.merge!(success: codes_for(codes, resp_type, 'success'))
         end
       end
 
-      def codes_for(codes, type)
-        codes.map { |c| method("property_#{type}_codes").call[c.to_sym] }.compact
+      def codes_for(codes, resp_type, code_type)
+        codes.map do |code|
+          YAML.load(File.read("config/#{resp_type}_#{code_type}_codes.yml")).values.first[code.to_s]
```

Comment:
And then you would need to make a relevant `[response_type]_[code_type]_codes.yml` file in config. This naming convention is what makes it autonomous